lms/models/application_settings.py

```diff
@@ -1,21 +1,54 @@
-from copy import deepcopy
+from sqlalchemy.ext.mutable import MutableDict
 
 
-class ApplicationSettings:
-    """Model for accessing and updating application settings."""
+class ApplicationSettings(MutableDict):
+    """
+    Model for accessing and updating application settings.
 
-    def __init__(self, data):
-        self.data = data
+    This is a dict subclass, but you should use the custom get() and set()
+    methods below to get and set values because they're safer and more
+    convenient:
+
+    application_settings.get("group", "key")
+    : will return None if *either* "group" or "key" is missing from the dict
+
+    application_settings.set("group", "key", "value")
+    : will create the sub-dict "group" for you if it's missing from the dict,
+      and will notify SQLAlchemy of your change so that it gets saved to the DB
```

Comment:
Some of this seems like it might be better as documentation for the specific methods below. Also cut down on the content here. Keep this to more of "to be safe use set() and get() detailed below".

Reply:
Moved these into the methods